### PR TITLE
Feature/67 historical bike count map

### LINKED ISSUE #67: Slide with bike count map + chart
- [x] Map
    - [x] Should have GFR network as "baselayer"
- [x] Chart
- [x] Title
- [x] Text content
- [x] Design
- [x] Back-End providing data
-- live data + GFR: Done
-- Todo: historical data
- [x] Connect to the API

---

**Live API:** https://data-mobility.brussels/bike/api/counts/
**Live API with location:** http://data-mobility.brussels/geoserver/bm_bike/wfs?service=wfs&version=1.1.0&request=GetFeature&srsName=EPSG:4326&outputFormat=json&typeName=bm_bike:rt_counting

### PR DESCRIPTION
**Pull request by Dakad**
Fixes #67 

## Description

 - Regroup the pages by the module's name
 - Each page is represented by it `index.js`
 - Expose only the index.js for the app.js
 - The `index.js` must export 2 functions :
   --` init()` : To run all required modules if this page is active
   -- `changeLanguage()` : Action to execute when a new language has been selected
   
## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/js/app.js [hunk 1]
```diff
@@ -101,15 +99,14 @@ const onLangSelectorClick = function(ev) {
   $('.lang_selector.active').removeClass('active');
   $this.addClass('active')
 
-  const path = window.location.pathname.substr(1).slice(0, -5);
-
-  console.log(lang, path);
+  // const path = window.location.pathname.substr(1).slice(0, -5);
 
-  Translation.updateLang(path, lang, updateChartLanguage);
+  console.log(lang);
```
Comment (jbelien) [inline src/js/app.js]: Please remove all `console.log()`

File: src/js/app.js [hunk 2]
```diff
@@ -101,15 +99,14 @@ const onLangSelectorClick = function(ev) {
...
 
+  Translation.updateLang(lang);
+  Translation.notifyAll(lang)
 }
 
-
 document.addEventListener('DOMContentLoaded', () => {
   console.info('DOM loaded');
```
Comment (jbelien) [inline src/js/app.js]: Please remove all `console.info()`

File: src/js/pages/bike-count/bike-count-per-year-chart.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import Chart from 'chart.js';
 // Our labels along the x-axis
-const years = [
+let years = [
```
Comment (jbelien) [inline src/js/pages/bike-count/bike-count-per-year-chart.js:3]: Why those years do not come from API ?

File: src/js/pages/bike-count/bike-count-per-year-chart.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ const years = [
   2017
 ];
 // For drawing the lines
-const count = [
+let count = [
```
Comment (jbelien) [inline src/js/pages/bike-count/bike-count-per-year-chart.js:26]: Why those data do not come from the API ?

File: src/js/pages/bike-count/bike-count-season-chart.js [hunk 5]
```diff
@@ -1,35 +1,70 @@
 import Chart from 'chart.js';
 
+// Our labels along the x-axis
+let years = [
```
Comment (jbelien) [inline src/js/pages/bike-count/bike-count-season-chart.js:4]: Why those years do not come from API ?

File: src/js/pages/bike-count/bike-map.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        'circle-color': '#2d3e71',
+        'circle-radius': 13
+      }
+    },
+  );
+};
+
+
+export function onChangeLanguage(translations) {
+  console.log(translations);
```
Comment (jbelien) [inline src/js/pages/bike-count/bike-map.js]: Please remove all `console.log()`

File: src/js/pages/commute/commute-to-work-chart.js [hunk 7]
```diff
@@ -1,26 +1,27 @@
...
-  const years = chartData.map(d => d.year);
+  const years = chartData.map(d => d.years);
   // For drawing the lines
-  const percentage = chartData.map(d => d.percentage);
+  const percentage = chartData.map(d => d.percentage_commuting_to_work);
 
-  const _ctx = document.getElementById(`js-canvas-commute`);
-  if (_ctx) {
-     commuteChart = new Chart(_ctx, {
+  console.log(years, percentage);
```
Comment (jbelien) [inline src/js/pages/commute/commute-to-work-chart.js]: Please remove all `console.log()`

File: src/js/pages/evolution/index.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  }
+}
+
+
+/**
+ *
+ * @param {string} lang  - The language selected
+ */
+export function changeLanguage(lang){
+  console.log(lang);
```
Comment (jbelien) [inline src/js/pages/evolution/index.js]: Please remove all `console.log()`

File: src/js/pages/services/service-map.js [hunk 9]
```diff
@@ -216,7 +213,7 @@ const showMapLayers = () => {
 
 
 
-export function onChangeLanguage(lang, translations) {
-  console.log(lang, translations);
+export function onChangeLanguage(translations) {
+  console.log(translations);
```
Comment (jbelien) [inline src/js/pages/services/service-map.js]: Please remove all `console.log()`

**Review by jbelien — CHANGES_REQUESTED** [review]

File: src/js/pages/bike-count/bike-count-per-year-chart.js (hunk 4 above)
Comment (Dakad) [reply src/js/pages/bike-count/bike-count-per-year-chart.js:26]: It's coming from the API. This variable is used as template for the dataset

**Review by jbelien — APPROVED** [review]

**Review by jbelien — APPROVED** [review]

File: src/assets/i18n/bike-count.json [hunk 10]
```diff
@@ -13,5 +13,10 @@
     "en": "January ; May ; September ; November",
     "fr": "Janvier ; Mai ; Septembre ; Novembre",
     "nl": "Januari ; Mei ; September ; November"
+  },
+  "graph_title" : {
+    "en": "Click on the blue circle on the map to get more info",
+    "fr": "Click sur le point bleu de la carte pour + info",
```
Comment (jbelien) [inline src/assets/i18n/bike-count.json]: "Cliquez sur un point bleu dans la carte pour avoir l'historique cet endroit."

File: src/assets/i18n/bike-count.json [hunk 11]
```diff
@@ -13,5 +13,10 @@
...
+    "nl": "Nah"
```
Comment (jbelien) [inline src/assets/i18n/bike-count.json]: Let's translate it right now ; otherwise we might forget!

File: src/assets/i18n/bike-count.json [hunk 12]
```diff
@@ -13,5 +13,10 @@
     "en": "January ; May ; September ; November",
     "fr": "Janvier ; Mai ; Septembre ; Novembre",
     "nl": "Januari ; Mei ; September ; November"
+  },
+  "graph_title" : {
+    "en": "Click on the blue circle on the map to get more info",
```
Comment (jbelien) [inline src/assets/i18n/bike-count.json]: "Click on a blue circle on the map to get the historical data of this place."

**Review by jbelien — CHANGES_REQUESTED** [review]

**Pull request merged**